### PR TITLE
Implement initial CI integrations

### LINKED ISSUE #4: Initialize base CI jobs
Import GitHub CI configuration, including:

- md-lint
- spellcheck
- actionlint
- python formatting and linting (from `instructlab/instructlab`)

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Resolves #4 

This PR implements the following CI integrations:
- **Dependabot** via a GitHub Action configuration
- **Markdown Linting** implemented via `Makefile` as well as a Hardened GitHub Action
- **Spellchecking** implemented via `tox` (with a `Makefile` entrypoint) as well as a Hardened GitHub Action
- **Action Linting** implemented via `Makefile` as well as a Hardened GitHub Action
- **Python Linting and Formatting** via `tox` using `ruff` and `isort` (with a `Makefile` entrypoint) as well as a Hardened GitHub Action

It also addresses Markdown and Python linting errors that are now being detected

### CODE REVIEW

**Review by 2chiefk — APPROVED** [review]

**Comment by russellb** [comment]
Let me know when this is ready for review.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    schedule:
+      interval: "daily"
+  - package-ecosystem: "pip"
+    directory: "/tools/maintainers"
+    schedule:
+      interval: "daily"
+  - package-ecosystem: "pip"
+    directory: "/tools/triage"
+    schedule:
+      interval: "daily"
```
Comment (bjhargrave) [inline .github/dependabot.yml:29]:
```suggestion
```

These lines don't apply in this repo.

**Review by bjhargrave — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: .github/dependabot.yml (hunk 1 above)
Comment (nathan-weinberg) [reply .github/dependabot.yml:29]: Ah good catch! Fixed

**Comment by nathan-weinberg** [comment]
> Let me know when this is ready for review.

This should be good to go - can't test the actions themselves in this PR but everything other than Dependabot can be run locally

**Comment by nathan-weinberg** [comment]
I can squash commits if desired, left them separated for easier reviewing since this is a fairly large PR

**Review by bjhargrave — APPROVED** [review]

**Review by russellb — APPROVED** [review]

**Pull request merged**